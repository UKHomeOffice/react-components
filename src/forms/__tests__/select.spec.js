import React from 'react'
import { render } from 'enzyme'
import {Select} from 'govuk-react-components';

describe('The Select Component', () => {

  it('should render correctly with no options', () => {
    expect(
      render(<Select label="test" name="test" options={[]} />)
    ).toMatchSnapshot()
  });

  it('should render correctly with multiple options', () => {
    expect(
      render(<Select label="test" name="test" options={[{ label: 'opt1-label', value: 'opt1' }, { label: 'opt2-label', value: 'opt2' }]} />)
    ).toMatchSnapshot()
  });


  it('should render correctly with multiple options and one selected', () => {
    expect(
      render(<Select label="test" name="test" value={'opt2'} onChange={() => { }} options={[{ label: 'opt1-label', value: 'opt1' }, { label: 'opt2-label', value: 'opt2' }]} />)
    ).toMatchSnapshot()
  });

  it('should render a hint when a hint is passed in', () => {
    expect(
      render(<Select label="test" name="test" options={[]} hint="Testing hint" />)
    ).toMatchSnapshot();
  });

  it('should render an error when an error is passed in', () => {
    expect(
      render(<Select label="test" name="test" options={[]} error="Testing error" />)
    ).toMatchSnapshot();
  });

})
