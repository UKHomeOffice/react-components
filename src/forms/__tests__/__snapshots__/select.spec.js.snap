// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Select Component should render a hint when a hint is passed in 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="test"
  >
    <span
      class="form-label-bold"
    >
      test
    </span>
    <span
      class="form-hint"
    >
      Testing hint
    </span>
  </label>
  <select
    class="form-control"
    id="test"
    name="test"
  >
    <option
      value=""
    >
      Select...
    </option>
  </select>
</div>
`;

exports[`The Select Component should render an error when an error is passed in 1`] = `
<div
  class="form-group form-group-error"
>
  <label
    class="form-label"
    for="test"
  >
    <span
      class="form-label-bold"
    >
      test
    </span>
    <span
      class="error-message"
    >
      Testing error
    </span>
  </label>
  <select
    class="form-control form-control-error"
    id="test"
    name="test"
  >
    <option
      value=""
    >
      Select...
    </option>
  </select>
</div>
`;

exports[`The Select Component should render correctly with multiple options 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="test"
  >
    <span
      class="form-label-bold"
    >
      test
    </span>
  </label>
  <select
    class="form-control"
    id="test"
    name="test"
  >
    <option
      value=""
    >
      Select...
    </option>
    <option
      value="opt1"
    >
      opt1-label
    </option>
    <option
      value="opt2"
    >
      opt2-label
    </option>
  </select>
</div>
`;

exports[`The Select Component should render correctly with multiple options and one selected 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="test"
  >
    <span
      class="form-label-bold"
    >
      test
    </span>
  </label>
  <select
    class="form-control"
    id="test"
    name="test"
  >
    <option
      value=""
    >
      Select...
    </option>
    <option
      value="opt1"
    >
      opt1-label
    </option>
    <option
      selected=""
      value="opt2"
    >
      opt2-label
    </option>
  </select>
</div>
`;

exports[`The Select Component should render correctly with no options 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="test"
  >
    <span
      class="form-label-bold"
    >
      test
    </span>
  </label>
  <select
    class="form-control"
    id="test"
    name="test"
  >
    <option
      value=""
    >
      Select...
    </option>
  </select>
</div>
`;
